handler/src/main/java/io/netty/handler/ssl/AbstractSniHandler.java

```diff
@@ -117,8 +121,50 @@ private static String extractSniHostname(ByteBuf in) {
         return null;
     }
 
+    protected final long handshakeTimeoutMillis;
+    private ScheduledFuture<?> timeoutFuture;
     private String hostname;
 
+    /**
+     * @param handshakeTimeoutMillis the handshake timeout in milliseconds
+     */
+    public AbstractSniHandler(long handshakeTimeoutMillis) {
+        this.handshakeTimeoutMillis = checkPositiveOrZero(handshakeTimeoutMillis, "handshakeTimeoutMillis");
+    }
+
+    public AbstractSniHandler() {
+        this(0L);
+    }
+
+    @Override
+    public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+        if (ctx.channel().isActive()) {
+            checkStartTimeout(ctx);
+        }
+    }
+
+    @Override
+    public void channelActive(ChannelHandlerContext ctx) throws Exception {
+        checkStartTimeout(ctx);
+    }
+
+    protected void checkStartTimeout(final ChannelHandlerContext ctx) {
```

Comment:
Any reasons why this is protected ?
```suggestion
    private void checkStartTimeout(final ChannelHandlerContext ctx) {
```

Reply:
nope